File: living-frontend/interfaces/posts/post.ts
import { User } from "interfaces/user/user";

export interface Post {
    id: number;
    content: string;
    created_at: string;
    user_id: number;
    user: User | null;
    likes: Array<Like>;
}



export interface Like {
    id: number;
    user_id: number;
    user: User | null;
    post_id: number;
    post: Post | null;
    created_at: string;
}

export interface Attachment {
    id: number;
    url: string;
    post_id: number;
    post: Post | null;
    attachment_type: string
}

export interface Group {
    id: number;
    name: string;
    owner_id: number;
    owner: User | null;
    description: string;
    photo_url: string;
    banner_url: string;
    created_at: string;
    updated_at: string;
    posts: Array<Post>;
    users: Array<User>;
}

File: living-backend/Models/Posts/Post.cs
using living_backend.Models.Groups;
using living_backend.Models.Users;
using System.ComponentModel.DataAnnotations.Schema;

namespace living_backend.Models.Posts;

public class Post
{
    public int Id { get; set; }
    [ForeignKey(nameof(User))]
    public int UserId { get; set; }
    [ForeignKey(nameof(Group))]
    public int? GroupId { get; set; }

    public string Content { get; set; }
    public int LikesCount { get; }
    public Access Access { get; set; }
    public DateTime CreatedAt { get; set; }

    public virtual User User { get; set; }
    public virtual Group Group { get; set; }
    public virtual ICollection<Attachment> Attachments { get; set; }
    public virtual ICollection<Like> Likes { get; set; }
}

public enum Access
{
    Public,
    Followers,
    Group,
    Private
}
